pkg/application/applyBlueprintSpecUseCase.go

```diff
@@ -0,0 +1,112 @@
+package application
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/cloudogu/k8s-blueprint-operator/pkg/domain"
+	"github.com/cloudogu/k8s-blueprint-operator/pkg/domainservice"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+type ApplyBlueprintSpecUseCase struct {
+	repo               domainservice.BlueprintSpecRepository
+	doguInstallUseCase doguInstallationUseCase
+}
+
+func NewApplyBlueprintSpecUseCase(
+	repo domainservice.BlueprintSpecRepository,
+	doguInstallUseCase doguInstallationUseCase,
+) *ApplyBlueprintSpecUseCase {
+	return &ApplyBlueprintSpecUseCase{
+		repo:               repo,
+		doguInstallUseCase: doguInstallUseCase,
+	}
+}
+
+func (useCase *ApplyBlueprintSpecUseCase) ApplyBlueprintSpec(ctx context.Context, blueprintId string) error {
+	blueprintSpec, err := useCase.repo.GetById(ctx, blueprintId)
+	if err != nil {
+		return fmt.Errorf("cannot load blueprint to apply blueprint spec: %w", err)
+	}
+
+	err = useCase.markInProgress(ctx, blueprintSpec)
+	if err != nil {
+		return err
+	}
+
+	//TODO: activate maintenance mode
+
+	applyError := useCase.doguInstallUseCase.ApplyDoguStates(ctx, blueprintId)
+	if applyError != nil {
+		err := useCase.MarkFailed(ctx, blueprintSpec, err)
+		if err != nil {
+			return err
+		}
+		return applyError
+	}
+
+	//TODO: set state to wait for health
+	//healthError := useCase.markWaitingForHealthyEcosystem(ctx, blueprintSpec)
+	//if healthError != nil {
+	//	return healthError
+	//}
+
+	//TODO: deactivate maintenance mode
+
+	//TODO: need to check ecosystem health here
+	//err = useCase.doguInstallUseCase.CheckDoguHealth(ctx, blueprintId)
+	//if err != nil {
+	//	return err
+	//}
```

Comment:
I know this comes with a later PR. So I guess, it's ok to leave these TODO comments here?

Reply:
Yes, i added the TODOs, so that it gets clear, where to add upcoming code and to make clear that this process is not complete yet. We just apply the dogu state without looking for health etc now.